aws_lambda_powertools/utilities/idempotency/persistence.py

```diff
@@ -0,0 +1,560 @@
+"""
+Persistence layers supporting idempotency
+"""
+
+import datetime
+import hashlib
+import json
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Optional, Union
+
+import boto3
+import jmespath
+from botocore.config import Config
+
+from .cache_dict import LRUDict
+from .exceptions import IdempotencyValidationerror, InvalidStatusError, ItemAlreadyExistsError, ItemNotFoundError
+
+logger = logging.getLogger(__name__)
+
+STATUS_CONSTANTS = {
+    "NOTEXISTING": "DOESNOTEXIST",
+    "INPROGRESS": "INPROGRESS",
+    "COMPLETED": "COMPLETED",
+    "EXPIRED": "EXPIRED",
```

Comment:
If we use TTL the record might be deleted by Dynamdob (or Redis). So i guess we could have a TTL record which is longer than the EXPIRED record?

Reply:
Yep, the TTL set in DDB or Redis needs to be >= the setting in the persistence instance. I'll make a note to add that to the documentation.